### ISSUE TITLE
s3: getSignedUrl is missing

### ISSUE THREAD

**Issue by TillaTheHun0** [cloudydeno/deno-aws_api#29]
I was looking to use [getSignedUrl](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html#getSignedUrl-property) for uploading directly to s3, but didn't see the method in the generated s3 service. I do see it on the [s3 service on the aws sdk referenced on main](https://github.com/aws/aws-sdk-js/blob/7eb7cae1cb01c4541f3687822f238a957e142409/lib/services/s3.js)

Am I looking in the wrong place? Thanks for your help!

**Comment by danopia** [comment]
Hey, thanks for writing!

High level, you're not missing anything here. The focus of this library is to provide clients to the AWS APIs, and pre-signing isn't actually an API call so it wasn't really in scope. This was previously brought up in #5

In that linked discussion, there is a link to an /x/ module just for pre-signing, check out the message here: https://github.com/cloudydeno/deno-aws_api/issues/5#issuecomment-927862540 and I also have an example of using it with the /x/aws_api credential loader. I used that pre-signing module in another project and it does the trick, albeit with a somewhat narrow API.

In summary, it would be interesting to work pre-signing methods into this module but there's nothing currently committed or in progress towards this feature

**Comment by TillaTheHun0** [comment]
Agh, I missed that issue. Apologies for the dupe. I'll try out the solutions proposed there. Feel free to close this.

**Comment by danopia** [comment]
All good! I'd still like to track common requests like these so that I can discover what additional features to work on. In this case, the big dependency for pre-signing is defining an internal API to build signed URLs. It would be worth doing if the other options are lacking for some reason or another.

Regardless, I haven't documented this sort of limitation properly, so either way I have something to do before closing this issue :)

**Comment by danopia** [comment]
:rocket: This is added in `v0.8.0`. There is a low-level presigning API and a high-level "extras" module for S3 presigning.

Example usages of both APIs are in this comment:
* https://github.com/cloudydeno/deno-aws_api/issues/5#issuecomment-1445245504
